kin-core/src/main/java/kin/core/AccountInfoRetriever.java

```diff
@@ -62,6 +75,74 @@ Balance getBalance(@NonNull String accountId) throws OperationFailedException {
         return balance;
     }
 
+    /**
+     * @param accountId the account ID to get all its transaction history
+     * @return the list of payments {@link PaymentInfo}
+     * @throws OperationFailedException or one of its subclasses
+     */
+    List<PaymentInfo> getPaymentsHistory(@NonNull String accountId) throws OperationFailedException {
+        List<PaymentInfo> payments = new ArrayList<>();
+        if (blockchainEvents != null) {
+            payments = getPaymentsHistory(server.transactions(), accountId);
+        }
+
+        return payments;
+    }
+
+    /**
+     * @param requestParams is an optional parameters to the request, such as accountId, cursor, limit, and order.
+     *                      If no account is given then using the current account if exist.
+     * @return the list of payments {@link PaymentInfo}
+     * @throws OperationFailedException or one of its subclasses
+     */
+    List<PaymentInfo> getPaymentsHistory(PaymentsHistoryRequestParams requestParams) throws OperationFailedException {
+        List<PaymentInfo> payments = new ArrayList<>();
+        TransactionsRequestBuilder transactionBuilder = server.transactions();
+        // Add all the optional parameters that the client supplied (if any supplied) to the request
+        if (requestParams.getLimit() > 0) {
+            transactionBuilder.limit(requestParams.getLimit());
+        }
+        if (!TextUtils.isEmpty(requestParams.getToken())) {
+            transactionBuilder.cursor(requestParams.getToken());
+        }
+        if (requestParams.getOrder() != null) {
+            transactionBuilder.order(RequestBuilder.Order.valueOf(requestParams.getOrder().name()));
+        }
+        if (blockchainEvents != null) {
+            payments = getPaymentsHistory(transactionBuilder, requestParams.getAccountId());
+        }
+        return payments;
+    }
+
+    private List<PaymentInfo> getPaymentsHistory(TransactionsRequestBuilder transactions, @NonNull String accountId) throws OperationFailedException {
+        List<PaymentInfo> payments = new ArrayList<>();
+        try {
+            Page<TransactionResponse> execute = transactions
+                    .forAccount(KeyPair.fromAccountId(accountId))
+                    .execute();
+            if (execute == null || execute.getRecords() == null) {
+                throw new AccountNotActivatedException(accountId);
```

Comment:
why it's `AccountNotActivatedException` in this case?
what does null response mean? I would expect general exception (data cannot be retrieved) like in getBalance/sendTransaction